Essay/#35 post-get self-made essay

Raelav opened the pull request:
Ветка выделена из #34 (Которая #17 задача). Нас интересуют последние 3 коммита по ним и смотри изменения.

Хотел определенные коммиты под задачу отправить - отправились все.
На счет последнего коммита "Получить все эссе по конкретному посту(новости)", там пока не уверен стоит ли выделять в отдельную задачу

Raelav committed:
Написание и правка эссе

Raelav committed:
Получить все группы, которым преподаватель давал задания

Raelav committed:
Получить все эссе по конкретному посту(новости)

gimrin commented on db-scripts/liquibase/sql/0001_InfoSync_ddl_initial.sql:
```diff
@@ -67,9 +67,11 @@ CREATE TABLE ifs_post_status
 (
     user_id BIGINT NOT NULL,
     post_id BIGINT NOT NULL,
+    essay_id BIGINT,
```
зачем здесь essay_id, если в в самой таблице essay есть и post_id и user_id?

gimrin committed:
Merge remote-tracking branch 'origin/#17_groups_info_for_teacher' into Essay/#35_post-get_self-made_essay

# Conflicts:
#	infosync-app/src/main/java/ru/urfu/infosync/dao/PostDao.java
#	infosync-app/src/main/java/ru/urfu/infosync/dao/PostStatusDao.java
#	infosync-app/src/main/java/ru/urfu/infosync/service/GroupService.java

gimrin commented on infosync-app/src/main/java/ru/urfu/infosync/service/GroupService.java:
```diff
@@ -33,4 +39,17 @@ public Integer createGroupOrGetId(String groupName) {
+    public List<Group> getGroupsRecommendedByTeacher(Integer teacherId) {
+
+        System.out.println("GroupService.getGroupsRecommendedByTeacher");
+        var x = postDao.getGroupsWithPostByTeacher(teacherId);
+        System.out.println(x);
+        System.out.println(groupDao.getGroupsNamesByIndexes(x));
```
тут явно что-то не доделанное осталось.
во-первых имя переменной, во вторых писать в консоль техническую инфу без контекста - такое себе.
Если тебе нужно что-то залоггировать, используй логгер.  slf4j например.

Raelav replied:
Это я как бы отладкой так некоторое время назад занимался. Теперь я овладел Debag'ом в idea, там все можно посмотреть

gimrin commented on infosync-app/src/main/java/ru/urfu/infosync/service/EssayService.java:
```diff
@@ -0,0 +1,41 @@
+    public boolean sendEssay(String text, Integer postId) {
+
+        var userId = userService.getIdCurrentUser();
+        if (text.isEmpty()) return false;
```
это можно делать сразу в контроллере.

gimrin commented on infosync-app/src/main/java/ru/urfu/infosync/model/PostStatus.java:
```diff
@@ -6,4 +6,11 @@
 public class PostStatus {
     Integer userId;
     Integer postId;
+    Integer essayId;
+
+    public PostStatus(Integer userId, Integer postId) {
+        this.userId = userId;
+        this.postId = postId;
+        this.essayId = null;
+    }
```
если нам не нужно эссе в статусе, это тоже нужно будет убрать.

gimrin commented on infosync-app/src/main/java/ru/urfu/infosync/controller/UserController.java:
```diff
@@ -13,13 +17,30 @@
+     * @return List of groups which the teacher recommended posts
+     */
+    @GetMapping(value = "/teacher/group", produces = APPLICATION_JSON_VALUE)
+    public List<Group> getGroupsRecommendedByTeacher(UsernamePasswordAuthenticationToken token) {
```
не понял как это к теме задачи относится, но ладно)

gimrin commented on infosync-app/src/main/java/ru/urfu/infosync/service/EssayService.java:
```diff
@@ -0,0 +1,41 @@
+        if (text.isEmpty()) return false;
+        var essayId = essayDao.sendEssay(text, postId, userId);
+        postStatusDao.addEssayId(userId, postId, essayId);
+        return text.equals(essayDao.getUserMadeEssay(postId, userId));
```
странная проверка) ты сохраняешь в базе эссе, потом читаешь и проверяешь с тем, что прислал юзер. тогда все ок, мы вроде как записали) 
по хорошему, если не было ошибок (exception) при записи в таблицу или ошибок валидации входных данных (у нас ее пока нет), то все, операция прошла. можно возвращать true.

Raelav committed:
-Исправил дублирование ссылки на эссе в разных таблицах
-Обобщил работу с исключаниями при доступе к БД (RegistrationResult -> InteractionBDResult)
-Удалил лишние и неиспользуемые части кода
-Добавил проверку на отсутсвие записей в БД при Гет-запросе предыдущего сделанного эсее

Raelav committed:
#35_repair of a crooked rebase
